src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityKMSClient.java

```diff
@@ -237,13 +237,12 @@ private static int getHeaderSize(byte[] bytes) {
      * @param bytes bytes to be checked
      */
     public static boolean isCiphertext(byte[] bytes) {
-        // Header size is currently always 0 for CMK encrypted docs. Expect at least one
-        // byte following the header that would have been encrypted.
-        // whenever header size is not 0, this should include a check that
-        // bytes.length > META_LENGTH + headerSize
+        // Header size is variable for CMK encrypted docs depending on whether
+        // the header is present. Expect at least one byte following the header
+        // that would have been encrypted.
         return bytes.length > DOCUMENT_HEADER_META_LENGTH
                 && bytes[0] == CURRENT_DOCUMENT_HEADER_VERSION && containsIroncoreMagic(bytes)
-                && getHeaderSize(bytes) == 0;
+                && getHeaderSize(bytes) >= 0;
```

Comment:
@skeet70 Just so you're aware, I had to change this because we haven't prioritized/worked on https://github.com/IronCoreLabs/tenant-security-client-java/issues/15. With this change we can at least prevent errors if someone tried to decrypt a document that was encrypted with the Node SDK. We can discuss in grooming/planning how far we want to take that other ticket and if/when to release this change.